spring-integration-core/src/main/java/org/springframework/integration/util/MessagingMethodInvokerHelper.java

```diff
@@ -267,8 +268,22 @@ else if (targetMethod instanceof String) {
 	private void prepareEvaluationContext() {
 		StandardEvaluationContext context = getEvaluationContext(false);
 		Class<?> targetType = AopUtils.getTargetClass(this.targetObject);
+
+		ReflectiveMethodResolver declaredMethodResolver = new ReflectiveMethodResolver() {
+
+			@Override
+			protected Method[] getMethods(Class<?> type) {
+				return Stream.of(type.getMethods(), type.getDeclaredMethods())
+						.flatMap(Stream::of)
+						.toArray(Method[]::new);
```

Comment:
Your first commit used this...

    return type.getDeclaredMethods();

which only returns the methods on the class (not those on the superclass(es)).

Your second commit provides a union of this and `getMethods()` - however, the public methods will be duplicated.